src/orquestra/vqa/algorithms/vqe.py

```diff
@@ -0,0 +1,323 @@
+    allocate_shots_uniformly,
+)
+
+
+class VQE:
+    def __init__(
+        self,
+        hamiltonian: PauliRepresentation,
+        optimizer: Optimizer,
+        ansatz: Ansatz,
+        estimation_method: EstimateExpectationValues,
+        grouping: EstimationPreprocessor,
+        shots_allocation: EstimationPreprocessor,
+        n_shots: Optional[int] = None,
+    ) -> None:
+        """Class providing an easy interface to work with VQE.
+
+        For new users, usage of "default" method is recommended.
+
+        Args:
+            hamiltonian: Cost Hamiltonian defining the problem.
+            optimizer: Optimizer used to find optimal parameters
+            ansatz: Ansatz defining what circuit will be used.
+            estimation_method: Method used for calculating expectation values of
+                the Hamiltonian.
+            grouping: Transforms list of estimation tasks by grouping and adding
+            context selection logic to the circuits
+            shots_allocation: Function allocate the shots for each task
+            n_shots: Number of shots to be used for evaluation of expectation values.
+                For simulation with exact expectation value it should be None.
+        """
+        self.hamiltonian = hamiltonian
+        self.optimizer = optimizer
+        self.ansatz = ansatz
+        self.estimation_method = estimation_method
+        self.grouping = grouping
+        self.shots_allocation = shots_allocation
+        self._n_shots = n_shots
+
+    @classmethod
+    def default(
+        cls,
+        hamiltonian: PauliRepresentation,
+        ansatz: Ansatz,
+        use_exact_expectation_values: bool = True,
+        grouping: str = "greedy",
+        shots_allocation: str = "proportional",
+        n_shots: Optional[int] = None,
+    ) -> "VQE":
+        """Creates a VQE object with some default settings:
+        - optimizer: L-BFGS-B optimizer (scipy implementation)
+
+        - ansatz: ansatz used for
+        - estimation method: either using exact expectation values
+            (only for simulation) or standard method of calculating expectation
+            values through averaging the results of measurements.
+
+        These can be later replaced using one of the `replace_*` methods.
+
+        Args:
+            hamiltonian: Cost Hamiltonian defining the problem.
+            use_exact_expectation_values: A flag indicating whether to use exact
+                calculation of the expectation values. This is possible only when
+                running on a simulator. Defaults to True.
+            grouping: name of the grouping function provided as a string. It only
+            accepts "greedy" and "individual" as an argument.
+            shots_allocation: name of the shots allocation function provided as a
+            string. It only accepts "proportional" and "individual" as an argument.
+            n_shots: If non-exact method for calculating expectation values is used,
```

Comment:
I would rephrase to:
"Specifies number of shots to be used for a given shots allocation method. If exact_expectation_values is true, it should be equal to `None`.